dbt/include/snowflake/macros/adapters.sql

```diff
@@ -42,11 +42,27 @@
   {%- set secure = config.get('secure', default=false) -%}
   {%- set copy_grants = config.get('copy_grants', default=false) -%}
   {%- set sql_header = config.get('sql_header', none) -%}
+  {% set existing_columns = adapter.get_columns_in_relation(relation) | map(attribute="name") | list %}
+  {% set model_columns = model.columns %}
 
+  {% set query_columns = get_columns_in_query(sql + ' limit 0') %}
```

Comment:
1. Let's avoid running this query if not `config.persist_column_docs()`, most likely by nesting it within the existing conditional below
2. The [`get_columns_in_query` macro](https://github.com/dbt-labs/dbt-core/blob/d5d16f01f4d18619ec4200728860dc64f5ce70c9/core/dbt/include/global_project/macros/adapters/columns.sql#L24) already handles adding `where false limit 0` + wrapping in a subquery
```suggestion
  {% set query_columns = get_columns_in_query(sql) %}
```

Reply:
Updated in 66312f7.